router/odp.go

```diff
@@ -1,25 +1,57 @@
 package router
 
 import (
+	"fmt"
+	"net"
+	"syscall"
+
 	"github.com/weaveworks/go-odp/odp"
 )
 
 // ODP admin functionality
 
-func CreateDatapath(dpname string) error {
+func CreateDatapath(dpname string) (err error, supported bool) {
 	dpif, err := odp.NewDpif()
 	if err != nil {
-		return err
+		if odp.IsKernelLacksODPError(err) {
+			return nil, false
+		}
+
+		return err, true
 	}
 
 	defer dpif.Close()
 
-	_, err = dpif.CreateDatapath(dpname)
+	dp, err := dpif.CreateDatapath(dpname)
 	if err != nil && !odp.IsDatapathNameAlreadyExistsError(err) {
-		return err
+		return err, true
+	}
+
+	// Pick an ephemeral port number to use in probing for vxlan
+	// support.
+	udpconn, err := net.ListenUDP("udp4", nil)
+	if err != nil {
+		return err, true
```

Comment:
It feels odd to return `true` here (even though it makes no difference in practice because the caller is always checking `err`)